### ISSUE TITLE
[urgent] Update LND and electrs due to a critical bug as soon as the fix is available

### ISSUE THREAD

**Issue by seberm** [mynodebtc/mynode#718]
**Describe the bug**
There is another bug in btcd which is causing LND to stuck and electrs to fail.

For more info, please see:
- btcd: https://github.com/btcsuite/btcd/issues/1906
- electrs: https://github.com/romanz/electrs/issues/783
- lnd: https://github.com/lightningnetwork/lnd/issues/7096
- https://twitter.com/c_otto83/status/1587400648366202880

**Comment by tlindi** [comment]
Electrum Server Status shows this error in logs:

Nov 01 07:11:46 mynoder systemd[1]: electrs.service: Failed with result 'exit-code'.
Nov 01 07:11:46 mynoder systemd[1]: electrs.service: Main process exited, code=exited, status=1/FAILURE
Nov 01 07:11:46 mynoder electrs[26807]:     2: receiving on an empty and disconnected channel
Nov 01 07:11:46 mynoder electrs[26807]:     1: failed to get block 000000000000000000070742427fa10ec3c66d006160155a704d9f56d090a3ea
Nov 01 07:11:46 mynoder electrs[26807]:     0: sync failed
Nov 01 07:11:46 mynoder electrs[26807]: Caused by:
Nov 01 07:11:46 mynoder electrs[26807]: Error: electrs failed
Nov 01 07:11:46 mynoder electrs[26807]: [2022-11-01T12:11:46.027Z INFO  electrs::db] closing DB at /mnt/hdd/mynode/electrs/bitcoin
Nov 01 07:11:46 mynoder electrs[26807]: note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Nov 01 07:11:46 mynoder electrs[26807]: thread 'p2p_recv' panicked at 'invalid message: OversizedVectorAllocation { requested: 12000072, max: 4000000 }', src/p2p.rs:211:62
Nov 01 07:11:46 mynoder electrs[26807]: [2022-11-01T12:11:46.002Z INFO  electrs::index] indexing 13 blocks: [761249..761261]
Nov 01 07:11:45 mynoder electrs[26807]: [2022-11-01T12:11:45.951Z INFO  electrs::chain] chain updated: tip=0000000000000000000640e114b9e6272b9688178273e193ef957e02fad1069f, height=761248
Nov 01 07:11:40 mynoder electrs[26807]: [2022-11-01T12:11:40.901Z INFO  electrs::chain] loading 761248 headers, tip=0000000000000000000640e114b9e6272b9688178273e193ef957e02fad1069f
Nov 01 07:11:35 mynoder electrs[26807]: [2022-11-01T12:11:35.996Z INFO  electrs::db] "/mnt/hdd/mynode/electrs/bitcoin": 158 SST files, 37.581945005 GB, 4.696187476 Grows
Nov 01 07:11:35 mynoder electrs[26807]: [2022-11-01T12:11:35.869Z INFO  electrs::server] serving Electrum RPC on 0.0.0.0:50001
Nov 01 07:11:35 mynoder electrs[26807]: [2022-11-01T12:11:35.869Z INFO  electrs::metrics::metrics_impl] serving Prometheus metrics on 127.0.0.1:4224
Nov 01 07:11:35 mynoder electrs[26807]: Starting electrs 0.9.3 on aarch64 linux with Config { network: Bitcoin, db_path: "/mnt/hdd/mynode/electrs/bitcoin", daemon_dir: "/home/bitcoin/.bitcoin", daemon_auth: CookieFile("/home/bitcoin/.bitcoin/.cookie"), daemon_rpc_addr: 127.0.0.1:8332, daemon_p2p_addr: 127.0.0.1:8333, electrum_rpc_addr: 0.0.0.0:50001, monitoring_addr: 127.0.0.1:4224, wait_duration: 15s, jsonrpc_timeout: 30s, index_batch_size: 10, index_lookup_limit: Some(200), reindex_last_blocks: 0, auto_reindex: true, ignore_mempool: false, sync_once: false, disable_electrum_rpc: false, server_banner: "Welcome to electrs 0.9.3 (Electrum Rust Server)!", args: [] }
Nov 01 07:11:35 mynoder systemd[1]: Started myNode Electrum Server.

**Comment by seberm** [comment]
The lnd with an emergency hotfix was just released:
- v0.15.4: https://github.com/lightningnetwork/lnd/releases/tag/v0.15.4-beta

**Comment by bucko13** [comment]
Any status update on this?

**Comment by quarreller** [comment]
I think(hope) tehelsper is working hard for the update.

**Comment by Z-and** [comment]
I'm having the same issue, was wondering why my LND was stuck in "waiting" and not synced with the blockchain. Will disabling the electrum wallet app resolve this?

**Comment by tlindi** [comment]
No, just sait for update or vendor work-a-round instruction here. IMHO

**Comment by tehelsper** [comment]
A new release with the fix is out now. It has:
- LND v0.15.4
- Electrs v0.9.9 for arm64 and amd64
- Better sig handling for LND allowing better app customization if this happens again and users need to upgrade early

**Comment by Z-and** [comment]
> A new release with the fix is out now. It has:
> 
> * LND v0.15.4
> * Electrs v0.9.9 for arm64 and amd64
> * Better sig handling for LND allowing better app customization if this happens again and users need to upgrade early

Awesome thank you!

Would this be beta_10-27-22_1666923526?

**Comment by tehelsper** [comment]
No, it's v0.3.04. You may need to click Check for Updates.

**Comment by Z-and** [comment]
> No, it's v0.3.04. You may need to click Check for Updates.

Downloading now

**Comment by jurglic** [comment]
I've upgraded to 0.3.04 from 0.3.03, but my electrs didn't get upgraded.
I can still see in electrs logs: "Starting electrs 0.9.3 on arm linux with Config".
And the issue remains, it doesn't start successfully.

**Comment by jurglic** [comment]
Looks like the electrs upgrade was only for 64bit and not for 32bit os?

**Comment by tehelsper** [comment]
Yes, the electrs binary was only made for amd64 and aarch64. If you are on the old 32-bit Raspberry Pi 4 OS, you can resolve the issue by re-flashing to the 64-bit image or by building electrs image yourself.

sudo mynode-install-extra electrs

The 32-bit binary will be updated in a future release.

**Comment by tehelsper** [comment]
The best solution is to upgrade to 64-bit to avoid future issues.

**Comment by tompro** [comment]
On pine64 (RockPro64) it updated electrs to v0.9.9 but when the server tries to start it crashes with following error:
```
myNode systemd[1]: electrs.service: Failed with result 'exit-code'.
myNode systemd[1]: electrs.service: Main process exited, code=exited, status=1/FAILURE
myNode electrs[21400]: /usr/bin/electrs: /lib/aarch64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /usr/bin/electrs)
myNode systemd[1]: Started myNode Electrum Server.

...
```

**Comment by tehelsper** [comment]
Oh, I think I may have unintentionally upgraded the RockPro64 image from Buster to Bullseye during the last image creation which has a different GLIBC version. Can you run these commands and compare?

```
admin@myNode:~$ ldd --version
ldd (Debian GLIBC 2.31-13+deb11u5) 2.31
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
admin@myNode:~$ cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
admin@myNode:~$ 
```

**Comment by tompro** [comment]
Nope seems to be buster:

```
admin@myNode:~$ ldd --version
ldd (Debian GLIBC 2.28-10+deb10u2) 2.28
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.

admin@myNode:~$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```
Screenshots to get a picture:

![image](https://user-images.githubusercontent.com/440368/200004090-673de0b8-0a79-4db0-93f0-eb3f3b04ac7f.png)

![image](https://user-images.githubusercontent.com/440368/200004530-92a69407-9061-47c7-90e9-ce83fe58dce1.png)

**Comment by KellsNode** [comment]
Updated 3.04 and still LND waiting.... Electrum stuck syncing at 99.96%

**Comment by tehelsper** [comment]
I just uploaded a new beta that should have fixes for Electrs on arm32 and aarch64.

**Comment by tompro** [comment]
Really appreciating your effort but I just updated to 3.05:

<img width="486" alt="image" src="https://user-images.githubusercontent.com/440368/201511932-4a2f0a0b-c1de-48a7-a55c-de3c6d1e84af.png">

The electrs issue for rockpro64 is still there. Just for clarity everything else is working just fine:

<img width="1117" alt="image" src="https://user-images.githubusercontent.com/440368/201511995-81b2ab88-383c-4b3b-a0ec-588e6f96efbb.png">

**Comment by ForteUnited** [comment]
Same issue here

**Comment by KellsNode** [comment]
Upgraded to 3.05, LND error:  [ERR] RPCS: [/lnrpc.Lightning/GetInfo]: waiting to start, RPC services not available

**Comment by tehelsper** [comment]
OK, @tompro and @ForteUnited, can you try running:

sudo mynode-install-extra rust
sudo mynode-install-extra electrs

That should get it working, at least until the next update which will overwrite electrs again. I built the latest electrs for aarch in docker for debian-buster but it must have still used a new glibc since I was running bullseye on that machine.

**Comment by tehelsper** [comment]
This may work as well and would be a lot faster. If it works, I can do an easy fix that re-uses the 64-bit image for RP4 on all aarch64 platforms.

```
sudo systemctl stop electrs
sudo wget https://github.com/mynodebtc/mynode/raw/master/rootfs/raspi4/usr/bin/electrs_arm64 -O /usr/bin/electrs
sudo systemctl restart electrs
```

**Comment by tompro** [comment]
Took your second approach and it works fine. Seems like the RP 64 bin can be reused. Thx for the help all green:

![image](https://user-images.githubusercontent.com/440368/201868377-774761bc-9a5b-4c86-b122-466f7720259a.png)

**Comment by KellsNode** [comment]
After upgrade, trying to generate new wallet address:

Generate New Address Failed
Error Code
500
Error Message
Unknown Error
API URL
./api/lnd/newaddress?type=1

What might be the problem?

**Comment by tehelsper** [comment]
Not sure. LND may not be running properly. Sounds like a different issue.

**Comment by ForteUnited** [comment]
> This may work as well and would be a lot faster. If it works, I can do an easy fix that re-uses the 64-bit image for RP4 on all aarch64 platforms.
> 
> ```
> sudo systemctl stop electrs
> sudo wget https://github.com/mynodebtc/mynode/raw/master/rootfs/raspi4/usr/bin/electrs_arm64 -O /usr/bin/electrs
> sudo systemctl restart electrs
> ```

This worked for me as well.

**Comment by samuelpj** [comment]
> This may work as well and would be a lot faster. If it works, I can do an easy fix that re-uses the 64-bit image for RP4 on all aarch64 platforms.
> 
> ```
> sudo systemctl stop electrs
> sudo wget https://github.com/mynodebtc/mynode/raw/master/rootfs/raspi4/usr/bin/electrs_arm64 -O /usr/bin/electrs
> sudo systemctl restart electrs
> ```

I tried running a recent build of "community edition" and had a similar error that was also resolved by these commands.
